---
title: "Downstream_RNASeq_AnalysisPlusFigures"
author: "ap2425"
date: "2025-04-14"
output:
  html_document:
    code_folding: hide
    code_download: true
    collapsed: yes
    highlight: tango
    number_sections: no
    theme: yeti
    toc: yes
    toc_float: yes
fontsize: 25pt
---

<style>
    body {
        text-align: justify;
    }

    .main-container {
        max-width: 1600px;
        margin-left: auto;
        margin-right: auto;
    }

    div.main-container {
        max-width: 1600px;
    }
</style>



## Set up

Setting up R, reading in file with sample info, setting colors for plotting.

``` {r setup, echo=TRUE, message=FALSE, results="hide", warning=FALSE, eval=TRUE}
require(knitr)
library(readr)
library(tidyverse)
library(ggplot2)
library(dplyr)
library(RColorBrewer)
library(kableExtra)
library(reshape2)
library(edgeR)
library(org.Mm.eg.db)
library(factoextra)
library(pheatmap)
library(RColorBrewer)
library(clusterProfiler)
```

## Read-in Counts

For the group annotations, the descriptions follow this pattern [ABC], where:  

A is genotype, M for mutant, W for wildtype  
B is sex, M for male, F for female  
C is tissue, B for brain, L for liver  

So MFB would be Mutant Female Brain Samples  

```{r eval=F}
night_count_df <- read.delim("./DarkOnset_counts.csv", sep = ',', header = T)
night_ann <- read.delim("./DarkOnset_groups.txt", sep = '\t', header = T)
night_ann$sample <- toupper(night_ann$sample %>% str_replace("s_",""))
night_ann$batch <- "new"
night_ann$batch[as.numeric(str_extract(night_ann$sample,"[0-9]+")) <= 1290] <- "old"
rownames(night_count_df) <- night_count_df$X
night_count_df <- night_count_df %>% select(-c(X))

morning_count_df <- read.delim("./LightOnset_counts.csv", sep = ',', header = T)
morning_ann <- read.delim("./LightOnset_groups.txt", sep = '\t', header = T)
morning_ann$sample <- toupper(morning_ann$sample %>% str_replace("s_",""))
morning_ann$batch <- "new"
morning_ann$batch[grepl("NR12",morning_ann$sample)] <- "old"
rownames(morning_count_df) <- morning_count_df$X
morning_count_df <- morning_count_df %>% select(-c(X))

genedata <- read.table("GRCm39.Ens_110.gene_data.txt",quote='',header=T,sep='\t',as.is = T)
```



## DarkOnset Analysis


### Differential Expression - Blocking

```{r eval=F}
samples <- nrow(night_ann)
normCounts <- data.frame(matrix(NA,nrow = length(night_count_df[,1]), ncol = length(samples)))
edger.rpkm <- data.frame(matrix(NA,nrow=length(night_count_df[,1]), ncol=length(samples)))

#Create DGEObject
y <- DGEList(counts = night_count_df, group = night_ann$group)

#Design model that will be used for DE comparisons
design <- model.matrix(~0+night_ann$group, data = y$samples)
colnames(design) <- levels(y$samples$group)

keep <- filterByExpr(y, design = design)
y <- y[keep, keep.lib.sizes=FALSE]


test_keep_pt <- length(which(keep))/length(night_count_df[,1])  #30% of genes kept after filtering by expression

y <- calcNormFactors(y)

#estimate common and tagwise dispersion
y <- estimateDisp(y,design)

#Grab count per million(cpm) values
normCounts <- cpm(y, normalized.lib.sizes = TRUE)
colnames(normCounts) <- paste("cpm.",night_ann$sample)
TempNormCounts <- normCounts

#create generalized linear model
fit <- glmQLFit(y,design)
```


```{r eval=F}
y <- DGEList(counts = night_count_df, group = night_ann$group)

#Design model that will be used for DE comparisons
treatment <- factor(night_ann$group)
batch <- factor(night_ann$batch)
blocked_design <- model.matrix(~batch + treatment)

keep <- filterByExpr(y, design = blocked_design)
y <- y[keep, keep.lib.sizes=FALSE]


test_keep_pt <- length(which(keep))/length(night_count_df[,1])  #30% of genes kept after filtering by expression

y <- calcNormFactors(y)

#estimate common and tagwise dispersion
y <- estimateDisp(y,blocked_design)

#Grab count per million(cpm) values
normCounts <- cpm(y, normalized.lib.sizes = TRUE)
colnames(normCounts) <- paste("cpm.",night_ann$sample)
TempNormCounts <- normCounts

#create generalized linear model
fit <- glmQLFit(y,blocked_design)

FL <- c(0,0,1,0,0,0,-1,0,0)
ML <- c(0,0,0,0,1,0,0,0,-1)
#FB <- c(0,0,1,0,0,0,-1,0,0)
MB <- c(0,0,0,1,0,0,0,-1,0)

res <- list()
res[[1]] <- glmQLFTest(fit, contrast = FL)
res[[2]] <- glmQLFTest(fit, contrast = ML)
res[[3]] <- glmQLFTest(fit, coef = 6)
res[[4]] <- glmQLFTest(fit, contrast = MB)

conts <- c("FL","ML","FB","MB")

summarytable <- data.frame(round(normCounts))
for(i in 1:length(conts))
{
  res[[i]]$table$padj <- p.adjust(res[[i]]$table$PValue, method = "BH")
  cur <- res[[i]]$table[,2]
  summarytable <- cbind(summarytable, cur)
  colnames(summarytable)[ncol(summarytable)] <- paste(conts[i],"logCPM")
}
for(i in 1:length(conts))
{
  cur <- res[[i]]$table[,c(1,4,5)]
  summarytable <- cbind(summarytable, cur)
  colnames(summarytable)[(ncol(summarytable)-2) :ncol(summarytable)] <- paste(c(paste(rep(conts[i],each=3), rep(c(".log2fc",".pval",".padj"), times = 1), sep = '')))
}

rownames(summarytable) <- rownames(y$counts)

summarytable.ann <- data.frame(ensid = rownames(summarytable), genedata[match(rownames(summarytable), genedata[,1]),c("Name","Chrom","Start","End","Strand","Biotype")], summarytable[,1:length(summarytable[1,])])  #add gene annotation info to summary table
rownames(summarytable.ann) <- paste(summarytable.ann$Name,"(",summarytable.ann$ensid,")") 
summarytable.ann.sort <- summarytable.ann[order(summarytable.ann[,"MB.padj"]),] #order by most significant adjusted pval

write.table(summarytable.ann.sort,"Blocked_Night_summary_genes.tsv",sep = '\t', quote = F, row.names = F)

#Find the DE genes
ups <- list()
dns <- list()

for(i in 1:length(conts))
{
  
  iv.ups<-summarytable.ann.sort[,paste(conts[i],".padj",sep='')] < .05 & summarytable.ann.sort[,paste(conts[i],".log2fc",sep='')] >= log2(1.5)
  iv.dn<-summarytable.ann.sort[,paste(conts[i],".padj",sep='')] < .05 & summarytable.ann.sort[,paste(conts[i],".log2fc",sep='')] <= -log2(1.5)
  
  ups[[i]] <- iv.ups
  dns[[i]] <- iv.dn
  write.table(summarytable.ann.sort[iv.ups,], paste("Blocked_Night_DE_",conts[i],".de.up.txt",sep=''), sep = '\t', quote =F, row.names = F)
  write.table(summarytable.ann.sort[iv.dn,], paste("Blocked_Night_DE_",conts[i],".de.dn.txt",sep=''), sep = '\t', quote =F, row.names = F)
  
}
```

## LightOnset Analysis

### Differential Expression - Blocking


```{r eval=F}
samples <- nrow(morning_ann)
normCounts <- data.frame(matrix(NA,nrow = length(morning_count_df[,1]), ncol = length(samples)))
edger.rpkm <- data.frame(matrix(NA,nrow=length(morning_count_df[,1]), ncol=length(samples)))
#tpm <- read.csv("../data/RSEM_TPM_table.csv", header = T, as.is = T)

#Create DGEObject
y <- DGEList(counts = morning_count_df, group = morning_ann$group)

#Design model that will be used for DE comparisons
design <- model.matrix(~0+morning_ann$group, data = y$samples)
colnames(design) <- levels(y$samples$group)

keep <- filterByExpr(y, design = design)
y <- y[keep, keep.lib.sizes=FALSE]


test_keep_pt <- length(which(keep))/length(morning_count_df[,1])  #30% of genes kept after filtering by expression

y <- calcNormFactors(y)

#estimate common and tagwise dispersion
y <- estimateDisp(y,design)

#Grab count per million(cpm) values
normCounts <- cpm(y, normalized.lib.sizes = TRUE)
colnames(normCounts) <- paste("cpm.",morning_ann$sample)
TempNormCounts <- normCounts

#create generalized linear model
fit <- glmQLFit(y,design)
```


```{r eval=F}
y <- DGEList(counts = morning_count_df, group = morning_ann$group)

#Design model that will be used for DE comparisons
treatment <- factor(morning_ann$group)
batch <- factor(morning_ann$batch)
blocked_design <- model.matrix(~batch + treatment)

keep <- filterByExpr(y, design = blocked_design)
y <- y[keep, keep.lib.sizes=FALSE]


test_keep_pt <- length(which(keep))/length(morning_count_df[,1])  #30% of genes kept after filtering by expression

y <- calcNormFactors(y)

#estimate common and tagwise dispersion
y <- estimateDisp(y,blocked_design)

#Grab count per million(cpm) values
normCounts <- cpm(y, normalized.lib.sizes = TRUE)
colnames(normCounts) <- paste("cpm.",morning_ann$sample)
TempNormCounts <- normCounts

#create generalized linear model
fit <- glmQLFit(y,blocked_design)

FL <- c(0,0,1,0,0,0,-1,0,0)
ML <- c(0,0,0,0,1,0,0,0,-1)
#FB <- c(0,0,1,0,0,0,-1,0,0)
MB <- c(0,0,0,1,0,0,0,-1,0)

res <- list()
res[[1]] <- glmQLFTest(fit, contrast = FL)
res[[2]] <- glmQLFTest(fit, contrast = ML)
res[[3]] <- glmQLFTest(fit, coef = 6)
res[[4]] <- glmQLFTest(fit, contrast = MB)

conts <- c("FL","ML","FB","MB")

summarytable <- data.frame(round(normCounts))
for(i in 1:length(conts))
{
  res[[i]]$table$padj <- p.adjust(res[[i]]$table$PValue, method = "BH")
  cur <- res[[i]]$table[,2]
  summarytable <- cbind(summarytable, cur)
  colnames(summarytable)[ncol(summarytable)] <- paste(conts[i],"logCPM")
}
for(i in 1:length(conts))
{
  cur <- res[[i]]$table[,c(1,4,5)]
  summarytable <- cbind(summarytable, cur)
  colnames(summarytable)[(ncol(summarytable)-2) :ncol(summarytable)] <- paste(c(paste(rep(conts[i],each=3), rep(c(".log2fc",".pval",".padj"), times = 1), sep = '')))
}

rownames(summarytable) <- rownames(y$counts)

summarytable.ann <- data.frame(ensid = rownames(summarytable), genedata[match(rownames(summarytable), genedata[,1]),c("Name","Chrom","Start","End","Strand","Biotype")], summarytable[,1:length(summarytable[1,])])  #add gene annotation info to summary table
rownames(summarytable.ann) <- paste(summarytable.ann$Name,"(",summarytable.ann$ensid,")") 
summarytable.ann.sort <- summarytable.ann[order(summarytable.ann[,"MB.padj"]),] #order by most significant adjusted pval

write.table(summarytable.ann.sort,"Blocked_Morning_summary_genes.tsv",sep = '\t', quote = F, row.names = F)

#Find the DE genes

ups <- list()
dns <- list()

for(i in 1:length(conts))
{
  
  iv.ups<-summarytable.ann.sort[,paste(conts[i],".padj",sep='')] < .05 & summarytable.ann.sort[,paste(conts[i],".log2fc",sep='')] >= log2(1.5)
  iv.dn<-summarytable.ann.sort[,paste(conts[i],".padj",sep='')] < .05 & summarytable.ann.sort[,paste(conts[i],".log2fc",sep='')] <= -log2(1.5)
  
  ups[[i]] <- iv.ups
  dns[[i]] <- iv.dn
  write.table(summarytable.ann.sort[iv.ups,], paste("Blocked_Morning_DE_",conts[i],".de.up.txt",sep=''), sep = '\t', quote =F, row.names = F)
  write.table(summarytable.ann.sort[iv.dn,], paste("Blocked_Morning_DE_",conts[i],".de.dn.txt",sep=''), sep = '\t', quote =F, row.names = F)
  
}

```

## GO-Term Enrichment

```{r eval=F}
blocked_night_fl_dn <- read.delim("Blocked_Night_DE_FL.de.dn.txt", sep = '\t')  
blocked_night_fl_up <- read.delim("Blocked_Night_DE_FL.de.up.txt", sep = '\t')
blocked_night_ml_dn <- read.delim("Blocked_Night_DE_ML.de.dn.txt", sep = '\t')  
blocked_night_ml_up <- read.delim("Blocked_Night_DE_ML.de.up.txt", sep = '\t')
blocked_night_fb_dn <- read.delim("Blocked_Night_DE_FB.de.dn.txt", sep = '\t')  
blocked_night_fb_up <- read.delim("Blocked_Night_DE_FB.de.up.txt", sep = '\t')
blocked_night_mb_dn <- read.delim("Blocked_Night_DE_MB.de.dn.txt", sep = '\t')  
blocked_night_mb_up <- read.delim("Blocked_Night_DE_MB.de.up.txt", sep = '\t') 

blocked_morning_fl_dn <- read.delim("Blocked_Morning_DE_FL.de.dn.txt", sep = '\t')  
blocked_morning_fl_up <- read.delim("Blocked_Morning_DE_FL.de.up.txt", sep = '\t')
blocked_morning_ml_dn <- read.delim("Blocked_Morning_DE_ML.de.dn.txt", sep = '\t')  
blocked_morning_ml_up <- read.delim("Blocked_Morning_DE_ML.de.up.txt", sep = '\t')
blocked_morning_fb_dn <- read.delim("Blocked_Morning_DE_FB.de.dn.txt", sep = '\t')  
blocked_morning_fb_up <- read.delim("Blocked_Morning_DE_FB.de.up.txt", sep = '\t')
blocked_morning_mb_dn <- read.delim("Blocked_Morning_DE_MB.de.dn.txt", sep = '\t')  
blocked_morning_mb_up <- read.delim("Blocked_Morning_DE_MB.de.up.txt", sep = '\t') 

des <- list(blocked_morning_fl_dn,blocked_morning_fl_up,blocked_morning_ml_dn, blocked_morning_ml_up, blocked_night_ml_dn, blocked_night_ml_up, blocked_night_fl_dn, blocked_night_fl_up)
de_names <- c("morning_fl_dn","morning_fl_up","morning_ml_dn","morning_ml_up","night_ml_dn","night_ml_up","night_fl_dn","night_fl_up")

i = 1
for(xi in des)
{
  if(nrow(xi) > 0)
  {
    cur_genes <- xi$ensid
    results <- enrichGO(gene = cur_genes, org.Mm.eg.db, keyType = "ENSEMBL", ont = "BP", pAdjustMethod = "BH",pvalueCutoff  = 0.1)
    results_df <- results@result
    results_df <- results_df %>% filter(p.adjust < 0.05)
    results_df <- arrange(results_df, p.adjust)
    if(nrow(results_df) > 0)
    {
      write_delim(results_df,paste0("GOBP_", de_names[i],"_results.txt"), delim = '\t', col_names = T)
      BP_colors_up=colorRampPalette(brewer.pal(n=9, name="Oranges"))(nrow(results_df))
      n <- min(nrow(results_df),10)
      #ggplot(results_df[1:10,], aes(y = reorder(ID, -1 * log10(p.adjust)) ,x = -1 * log10(p.adjust))) + geom_col(aes(fill = -1 * log10(p.adjust)), color = "black") + ylab("GOTerm") + scale_fill_gradientn(colors = BP_colors_up) + ggtitle(paste("GOBP_", de_names[i])) + geom_vline(xintercept = -1 * log10(0.05), color = "black")
      ggplot(results_df[1:n,], aes(y = reorder(ID, -1 * log10(p.adjust)) ,x = -1 * log10(p.adjust))) + geom_col(color = "black") + ylab("GOTerm") + ggtitle(paste("GOBP_", de_names[i])) + geom_vline(xintercept = -1 * log10(0.05), color = "black")
      ggsave(paste0("GOBP_", de_names[i],"_barplot.png"), height = 10, width = 10)
    }
  }
  i = i + 1
}


i = 1
for(xi in des)
{
  if(nrow(xi) > 0)
  {
    cur_genes <- xi$ensid
    results <- enrichGO(gene = cur_genes, org.Mm.eg.db, keyType = "ENSEMBL", ont = "MF", pAdjustMethod = "BH",pvalueCutoff  = 0.1)
    results_df <- results@result
    results_df <- results_df %>% filter(p.adjust < 0.05)
    results_df <- arrange(results_df, p.adjust)
    if(nrow(results_df) > 0)
    {
      write_delim(results_df,paste0("GOMF_", de_names[i],"_results.txt"), delim = '\t', col_names = T)
      BP_colors_up=colorRampPalette(brewer.pal(n=9, name="Blues"))(nrow(results_df))
      n <- min(nrow(results_df),10)
      #ggplot(results_df[1:10,], aes(y = reorder(ID, -1 * log10(p.adjust)) ,x = -1 * log10(p.adjust))) + geom_col(aes(fill = -1 * log10(p.adjust)), color = "black") + ylab("GOTerm") + scale_fill_gradientn(colors = BP_colors_up) + ggtitle(paste("GOBP_", de_names[i])) + geom_vline(xintercept = -1 * log10(0.05), color = "black")
      ggplot(results_df[1:n,], aes(y = reorder(ID, -1 * log10(p.adjust)) ,x = -1 * log10(p.adjust))) + geom_col(color = "black") + ylab("GOTerm") + ggtitle(paste("GOMF_", de_names[i])) + geom_vline(xintercept = -1 * log10(0.05), color = "black")
      ggsave(paste0("GOMF_", de_names[i],"_barplot.png"), height = 10, width = 10)
    }
  }
  i = i + 1
}


i = 1
for(xi in des)
{
  if(nrow(xi) > 0)
  {
    cur_genes <- xi$ensid
    results <- enrichGO(gene = cur_genes, org.Mm.eg.db, keyType = "ENSEMBL", ont = "CC", pAdjustMethod = "BH",pvalueCutoff  = 0.1)
    if ((is.null(results)) == F)
    {
    results_df <- results@result
    results_df <- results_df %>% filter(p.adjust < 0.05)
    results_df <- arrange(results_df, p.adjust)
    if(nrow(results_df) > 0)
    {
      write_delim(results_df,paste0("GOCC_", de_names[i],"_results.txt"), delim = '\t', col_names = T)
      BP_colors_up=colorRampPalette(brewer.pal(n=9, name="Greens"))(nrow(results_df))
      n <- min(nrow(results_df),10)
      #ggplot(results_df[1:10,], aes(y = reorder(ID, -1 * log10(p.adjust)) ,x = -1 * log10(p.adjust))) + geom_col(aes(fill = -1 * log10(p.adjust)), color = "black") + ylab("GOTerm") + scale_fill_gradientn(colors = BP_colors_up) + ggtitle(paste("GOBP_", de_names[i])) + geom_vline(xintercept = -1 * log10(0.05), color = "black")
      ggplot(results_df[1:n,], aes(y = reorder(ID, -1 * log10(p.adjust)) ,x = -1 * log10(p.adjust))) + geom_col(color = "black") + ylab("GOTerm") + ggtitle(paste("GOCC_", de_names[i])) + geom_vline(xintercept = -1 * log10(0.05), color = "black")
      ggsave(paste0("GOCC_", de_names[i],"_barplot.png"), height = 10, width = 10)
    }
    }
  }
  i = i + 1
}
```

All DE genes combined:

```{r eval=F}
all_blocked_de <- unique(c(blocked_night_fl_dn$ensid,blocked_night_fl_up$ensid,blocked_night_ml_dn$ensid,blocked_night_ml_up$ensid,blocked_night_fb_dn$ensid,blocked_night_fb_up$ensid,blocked_night_mb_dn$ensid,blocked_night_mb_up$ensid,blocked_morning_fl_dn$ensid,blocked_morning_fl_up$ensid,blocked_morning_ml_dn$ensid,blocked_morning_ml_up$ensid,blocked_morning_fb_dn$ensid,blocked_morning_fb_up$ensid,blocked_morning_mb_dn$ensid,blocked_morning_mb_up$ensid))

results <- enrichGO(gene = all_blocked_de, org.Mm.eg.db, keyType = "ENSEMBL", ont = "BP", pAdjustMethod = "BH",pvalueCutoff  = 0.1)
results_bp <- results@result
results_bp <- results_bp %>% filter(p.adjust < 0.05)
results_bp <- arrange(results_bp, p.adjust)
write_delim(results_bp,"EnrichedGOTerms_BP_AllBlockedDEGenes.txt", delim = '\t', col_names = T)

results <- enrichGO(gene = all_blocked_de, org.Mm.eg.db, keyType = "ENSEMBL", ont = "MF", pAdjustMethod = "BH",pvalueCutoff  = 0.1)
results_mf <- results@result
results_mf <- results_mf %>% filter(p.adjust < 0.05)
results_mf <- arrange(results_mf, p.adjust)
write_delim(results_mf,"EnrichedGOTerms_MF_AllBlockedDEGenes.txt", delim = '\t', col_names = T)

results <- enrichGO(gene = all_blocked_de, org.Mm.eg.db, keyType = "ENSEMBL", ont = "CC", pAdjustMethod = "BH",pvalueCutoff  = 0.1)
results_cc <- results@result
results_cc <- results_cc %>% filter(p.adjust < 0.05)
results_cc <- arrange(results_cc, p.adjust)
write_delim(results_cc,"EnrichedGOTerms_CC_AllBlockedDEGenes.txt", delim = '\t', col_names = T)
```



